streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractStream.java

```diff
@@ -31,6 +31,8 @@
     protected final Set<String> sourceNodes;
 
     public AbstractStream(KStreamBuilder topology, String name, Set<String> sourceNodes) {
+        assert sourceNodes != null && !sourceNodes.isEmpty() : "parameter <sourceNodes> must not be null or empty";
```

Comment:
Instead of adding the assertion, could we use exception here in case it really happens, as 

```
"parameter ... is null or empty, which should not happen". 
```

And then could we add some unit tests that validate such exceptions are never thrown in different combination of operator cases?